ccl/changefeedccl: TestChangefeedWithNoDistributionStrategy failed

ccl/changefeedccl.TestChangefeedWithNoDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/acd782ea-c80f-4c0b-a0c5-d65d3183b37f?testReportRun=4&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [455b16592df7d8efd121b3ba1256fb477e227564](https://github.com/cockroachdb/cockroach/commits/455b16592df7d8efd121b3ba1256fb477e227564):


```
=== RUN   TestChangefeedWithNoDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedWithNoDistributionStrategy3469676430
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    changefeed_dist_test.go:229: found partitions: [{1 /Tenant/10/Table/104/1{-/2}, /Tenant/10/Table/104/1/{39-40}} {2 /Tenant/10/Table/104/1/{2-4}} {3 /Tenant/10/Table/104/1/{4-8}} {4 /Tenant/10/Table/104/1/{8-16}} {5 /Tenant/10/Table/104/1/{16-32}} {6 /Tenant/10/Table/104/1/3{2-9}, /Tenant/10/Table/104/{1/40-2}}]
    changefeed_dist_test.go:338: range counts: [3 2 4 8 16 31 0 0]
    changefeed_dist_test.go:370: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:370
        	Error:      	Should be true
        	Test:       	TestChangefeedWithNoDistributionStrategy
        	Messages:   	unexpected counts [3 2 4 8 16 31 0 0], partitions: [{1 /Tenant/10/Table/104/1{-/2}, /Tenant/10/Table/104/1/{39-40}} {2 /Tenant/10/Table/104/1/{2-4}} {3 /Tenant/10/Table/104/1/{4-8}} {4 /Tenant/10/Table/104/1/{8-16}} {5 /Tenant/10/Table/104/1/{16-32}} {6 /Tenant/10/Table/104/1/3{2-9}, /Tenant/10/Table/104/{1/40-2}}]
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedWithNoDistributionStrategy3469676430
--- FAIL: TestChangefeedWithNoDistributionStrategy (149.87s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=4</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedWithNoDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-36705

ccl/changefeedccl.TestChangefeedWithNoDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/cc97cb67-31b1-4199-a861-c906ae9aa994?testReportRun=10&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [067e48d29b9093038f6fcf2074cd761ffdcd4fe2](https://github.com/cockroachdb/cockroach/commits/067e48d29b9093038f6fcf2074cd761ffdcd4fe2):


```
=== RUN   TestChangefeedWithNoDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedWithNoDistributionStrategy954441901
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:157: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
    changefeed_dist_test.go:229: found partitions: [{1 /Tenant/10/Table/104/1{-/2}} {2 /Tenant/10/Table/104/1/{2-4}} {3 /Tenant/10/Table/104/1/{4-8}} {4 /Tenant/10/Table/104/1/{8-16}} {5 /Tenant/10/Table/104/1/{16-32}} {6 /Tenant/10/Table/104/1/{32-50}, /Tenant/10/Table/104/{1/51-2}} {8 /Tenant/10/Table/104/1/5{0-1}}]
    changefeed_dist_test.go:338: range counts: [2 2 4 8 16 31 0 1]
    changefeed_dist_test.go:370: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:370
        	Error:      	Should be true
        	Test:       	TestChangefeedWithNoDistributionStrategy
        	Messages:   	unexpected counts [2 2 4 8 16 31 0 1], partitions: [{1 /Tenant/10/Table/104/1{-/2}} {2 /Tenant/10/Table/104/1/{2-4}} {3 /Tenant/10/Table/104/1/{4-8}} {4 /Tenant/10/Table/104/1/{8-16}} {5 /Tenant/10/Table/104/1/{16-32}} {6 /Tenant/10/Table/104/1/{32-50}, /Tenant/10/Table/104/{1/51-2}} {8 /Tenant/10/Table/104/1/5{0-1}}]
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedWithNoDistributionStrategy954441901
--- FAIL: TestChangefeedWithNoDistributionStrategy (106.03s)
```

Parameters:
 - <code>=true</code>
 - <code>attempt=1</code>
 - <code>run=10</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedWithNoDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

ccl/changefeedccl.TestChangefeedWithNoDistributionStrategy [failed](https://tanzanite.cluster.engflow.com/invocations/default/6a0e88c2-4e81-43cc-9fcf-028fdeffcac4?testReportRun=7&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on master @ [0b6de9c809f8a4df2ba943a8c9dd023adb03b01d](https://github.com/cockroachdb/cockroach/commits/0b6de9c809f8a4df2ba943a8c9dd023adb03b01d):


```
=== RUN   TestChangefeedWithNoDistributionStrategy
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestChangefeedWithNoDistributionStrategy2760564705
    test_log_scope.go:81: use -show-logs to present logs inline
    changefeed_dist_test.go:229: found partitions: [{1 /Table/104/1{-/2}} {2 /Table/104/1/{2-4}} {3 /Table/104/1/{4-8}, /Table/104/{1/63-2}} {4 /Table/104/1/{8-16}} {5 /Table/104/1/{16-32}} {6 /Table/104/1/{32-63}}]
    changefeed_dist_test.go:338: range counts: [2 2 5 8 16 31 0 0]
    changefeed_dist_test.go:370: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/changefeed_dist_test.go:370
        	Error:      	Should be true
        	Test:       	TestChangefeedWithNoDistributionStrategy
        	Messages:   	unexpected counts [2 2 5 8 16 31 0 0], partitions: [{1 /Table/104/1{-/2}} {2 /Table/104/1/{2-4}} {3 /Table/104/1/{4-8}, /Table/104/{1/63-2}} {4 /Table/104/1/{8-16}} {5 /Table/104/1/{16-32}} {6 /Table/104/1/{32-63}}]
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestChangefeedWithNoDistributionStrategy2760564705
--- FAIL: TestChangefeedWithNoDistributionStrategy (88.94s)
```

Parameters:
 - <code>=true</code>
 - <code>attempt=1</code>
 - <code>run=7</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedWithNoDistributionStrategy.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/ccl/changefeedccl/changefeed_dist_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_dist_test.go
@@ -296,7 +296,11 @@ func newRangeDistributionTester(
 
 	// Distribute the leases exponentially across the first 5 nodes.
 	for i := 0; i < 64; i += 1 {
-		nodeID := int(math.Floor(math.Log2(float64(i)))) + 1
+		nodeID := 1
+		// Avoid log(0).
+		if i != 0 {
+			nodeID = int(math.Floor(math.Log2(float64(i)))) + 1
+		}
 		cmd := fmt.Sprintf(`ALTER TABLE x EXPERIMENTAL_RELOCATE VALUES (ARRAY[%d], %d)`,
 			nodeID, i,
 		)

